### ISSUE TITLE
Right prompt overlaps left prompt

### ISSUE THREAD

**Issue by romkatv** [Powerlevel9k/powerlevel9k#1306]
To reproduce:

```zsh
docker run -e LANG=C.UTF-8 -e LC_ALL=C.UTF-8 -e TERM=$TERM -it --rm ubuntu bash -uexc '
  cd
  apt update && apt install -y zsh git
  git clone https://github.com/bhilburn/powerlevel9k.git
  echo "
    POWERLEVEL9K_PROMPT_ON_NEWLINE=true
    POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(custom_left)
    POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS=(custom_right)
    typeset POWERLEVEL9K_CUSTOM_{LEFT,RIGHT}=\"echo \${(pl.\$((COLUMNS/2))..-.)}\"
    POWERLEVEL9K_CUSTOM_LEFT_BACKGROUND=red
    POWERLEVEL9K_CUSTOM_RIGHT_BACKGROUND=blue
    source ~/powerlevel9k/powerlevel9k.zsh-theme" >~/.zshrc
  exec zsh'
```

Actual:

![actual](https://user-images.githubusercontent.com/1282067/59246190-e249be00-8c1b-11e9-8b83-0005339a3739.png)

Expected:

![expected](https://user-images.githubusercontent.com/1282067/59246197-e70e7200-8c1b-11e9-9f16-0560cb63c7d6.png)

Normally, ZSH doesn't display right prompt if it would overlap left prompt. If you type the following command, you won't see right prompt.

```zsh
PROMPT="${(pl.$((COLUMNS/2))..-.)}> " RPROMPT="<${(pl.$((COLUMNS/2))..-.)}" zsh -df
```

Now, if you make your terminal wider, right prompt will appear.

If you remove `POWERLEVEL9K_PROMPT_ON_NEWLINE=true` from the p9k config above, right prompt won't be displayed, as expected.

```zsh
docker run -e LANG=C.UTF-8 -e LC_ALL=C.UTF-8 -e TERM=$TERM -it --rm ubuntu bash -uexc '
  cd
  apt update && apt install -y zsh git
  git clone https://github.com/bhilburn/powerlevel9k.git
  echo "
    POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(custom_left)
    POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS=(custom_right)
    typeset POWERLEVEL9K_CUSTOM_{LEFT,RIGHT}=\"echo \${(pl.\$((COLUMNS/2))..-.)}\"
    POWERLEVEL9K_CUSTOM_LEFT_BACKGROUND=red
    POWERLEVEL9K_CUSTOM_RIGHT_BACKGROUND=blue
    source ~/powerlevel9k/powerlevel9k.zsh-theme" >~/.zshrc
  exec zsh'
```

![one-line-prompt](https://user-images.githubusercontent.com/1282067/59246626-28ebe800-8c1d-11e9-8484-ca32329f6883.png)

**Comment by laggardkernel** [comment]
There's no builtin support for putting right prompt on the same line with left prompt.

https://github.com/bhilburn/powerlevel9k/blob/3057e8fc07374e9c580790a023cb85998ca91fc8/powerlevel9k.zsh-theme#L1823-L1827

You can't expect everything works perfectly in an "ugly workaround", right?

**Comment by romkatv** [comment]
> There's no builtin support for putting right prompt on the same line with left prompt.

True. There is also no builtin support for Git status, command execution time, nvm or pretty much anything Powerlevel9k provides. If everything was builtin in ZSH, there would be no reason for Powerlevel9k to exist.

> You can't expect everything works perfectly in an "ugly workaround", right?

Of course I expect the _current implementation_ to do exactly what it does. Sometimes, what the current implementation does is not what users expect or desire. We call them bugs.

I cannot speak for all users but to me the current behavior is neither expected nor desired. I gather from your comment that you share this opinion. If your point is that fixing the bug is hard, I wholeheartedly agree. It's possible though. I'm not at liberty to point to the solution on this forum. You likely can guess where it can be found.

**Comment by dritter** [comment]
Fixed on `next` with #1335 

Thanks @romkatv for the Bug reports.
